src/mongo/db/storage/hse/src/hse_durability_manager.cpp

```diff
@@ -123,19 +133,93 @@ void KVDBDurabilityManager::waitUntilDurable() {
         return;
 
     stdx::unique_lock<stdx::mutex> lk(_syncMutex);
-    const auto waitingFor = _numSyncs;
-
-    _syncDoneCV.wait(lk, [&] { return _shuttingDown.load() || (_numSyncs > waitingFor + 1); });
+    const auto waitingFor = _numSyncs + 1;
```

Comment:
not  "waiting for" anymore, would change to waitFor since this is updated already to value to "wait for".